src/server/scripts/Outland/BlackTemple/boss_shade_of_akama.cpp

```diff
@@ -1131,37 +1127,149 @@ class npc_ashtongue_spiritbinder : public CreatureScript
             {
-                if (HealthBelowPct(25))
-                {
-                    DoCast(me, SPELL_SPIRIT_MEND);
-                    spiritMend = true;
-                }
+                case ACTION_BROKEN_SPECIAL:
+                    Talk(SAY_BROKEN_SPECIAL);
+                    break;
+                case ACTION_BROKEN_HAIL:
+                    me->setFaction(FACTION_FRIENDLY);
+                    Talk(SAY_BROKEN_HAIL);
+                    break;
+                case ACTION_BROKEN_EMOTE:
+                    me->SetByteValue(UNIT_FIELD_BYTES_1, UNIT_BYTES_1_OFFSET_STAND_STATE, UNIT_STAND_STATE_KNEEL);
+                    break;
+                default:
+                    break;
             }
+        }
+
+    private:
+        InstanceScript* _instance;
+    };
+
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return GetInstanceAI<npc_ashtongue_brokenAI>(creature);
+    }
+};
+
+class spell_shade_soul_channel_serverside : public SpellScriptLoader
+{
+public:
+    spell_shade_soul_channel_serverside() : SpellScriptLoader("spell_shade_soul_channel_serverside") { }
+
+    class spell_shade_soul_channel_serverside_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_shade_soul_channel_serverside_AuraScript);
+
+        bool Validate(SpellInfo const* /*spell*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_SHADE_SOUL_CHANNEL_2))
+                return false;
+            return true;
+        }
+
+        void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            GetTarget()->RemoveAuraFromStack(SPELL_SHADE_SOUL_CHANNEL_2);
+        }
+
+        void Register() override
+        {
+            AfterEffectRemove += AuraEffectRemoveFn(spell_shade_soul_channel_serverside_AuraScript::OnRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_shade_soul_channel_serverside_AuraScript();
+    }
+};
+
+class spell_shade_soul_channel : public SpellScriptLoader
+{
+public:
+    spell_shade_soul_channel() : SpellScriptLoader("spell_shade_soul_channel") { }
 
-            if (!chainHeal)
+    class spell_shade_soul_channel_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_shade_soul_channel_AuraScript);
+
+        void UpdateSpeed()
+        {
+            Unit* target = GetTarget();
+            if (target->GetAuraCount(SPELL_SHADE_SOUL_CHANNEL_2) <= 3)
             {
-                if (HealthBelowPct(40))
-                {
-                    DoCast(me, SPELL_CHAIN_HEAL);
-                    chainHeal = true;
-                }
+                float moveSpeed = (2.0f - (0.6f * target->GetAuraCount(SPELL_SHADE_SOUL_CHANNEL_2)));
+                target->SetSpeedRate(MOVE_WALK, moveSpeed / 2.5f);
+                target->SetSpeedRate(MOVE_RUN, (moveSpeed * 2) / 7);
```

Comment:
Also, this limit to a minimum velocity of 0.1f breaks the movement: https://github.com/TrinityCore/TrinityCore/blob/6.x/src/server/game/Movement/Spline/MoveSpline.cpp#L209